Logging/.CSProject/Logger/Logger.cs

```diff
@@ -14,6 +15,33 @@ namespace Anvil.CSharp.Logging
     /// </summary>
     public readonly struct Logger : ILogger
     {
+        // NOTE: This is a duplicate of Anvil.CSharp.Reflection.TypeExtension.GetReadableName
+        // because the logging namespace is isolated from the rest of Anvil, and thus cannot access that extension
+        private static readonly Type s_NullableType = typeof(Nullable<>);
+        private static string GetReadableName(Type type)
+        {
+            if (!type.IsGenericType)
+            {
+                return type.Name;
+            }
+
+            if (type.GetGenericTypeDefinition() == s_NullableType)
+            {
+                return $"{GetReadableName(type.GenericTypeArguments[0])}?";
+            }
+
+            // Remove the generic type count indicator (`n) from the type name
+            // Avoid having to calculate the number of digits in the generic type count by assuming it's under 100
+            int removeCount = 1 + (type.GenericTypeArguments.Length < 10 ? 1 : 2);
+            string name = type.Name[..^removeCount];
+
+            string genericTypeNames = type.GenericTypeArguments
+                .Select(GetReadableName)
+                .Aggregate((a, b) => $"{a}, {b}");
```

Comment:
Lots of string allocations concatenating like this.

Fewer allocs:
```suggestion
            IEnumerable<string> readableArgs = type.GenericTypeArguments
                .Select(GetReadableName);
             string genericTypeNames = string.Join(readableArgs, ", ");
```

If we wanted to take it further we'd pass a `StringBuilder` through the recursion but that's probably excessive until we prove that logger is hammering the GC.